### PR TITLE
chore(sdk): Define helper trait `MaybeCompact`

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/paradigmxyz/reth/issues/12454

### CODE REVIEW

File: crates/primitives-traits/src/block/body.rs [hunk 1]
```diff
@@ -28,10 +27,15 @@ pub trait BlockBody:
     + InMemorySize
     + MaybeSerde
 {
+    /// Uncle block header.
+    type Header: BlockHeader + 'static;
+
```
Comment (mattsse) [inline crates/primitives-traits/src/block/body.rs:32]: this does not belong in this pr

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/primitives-traits/src/block/body.rs (hunk 1 above)
Comment (emhane) [reply crates/primitives-traits/src/block/body.rs:32]: if I restrict the bound to just `'static` is that ok? that way the `Compact` trait bound can properly be applied on `FullBlockBody`.

Comment (mattsse) [reply crates/primitives-traits/src/block/body.rs:32]: unclear why this is introduced, seems out of scope for this pr

**Comment by emhane** [comment]
re-request review @mattsse

**Comment by emhane** [comment]
ping @mattsse

File: crates/primitives-traits/Cargo.toml [hunk 2]
```diff
@@ -92,4 +92,5 @@ serde = [
     "revm-primitives/serde",
     "roaring/serde",
 	"revm-primitives/serde",
-]
\ No newline at end of file
+]
+reth-codec = []
```
Comment (mattsse) [inline crates/primitives-traits/Cargo.toml:96]: why is this a feature if reth-codecs dep is non optional?

Comment (emhane) [reply crates/primitives-traits/Cargo.toml:96]: will make the dep optional. unless the `reth-codec` feature is enabled for `reth-primitives`, none of the `reth-primitives` types will impl their corresponding `Full`- trait.

**Review by mattsse — APPROVED** [review]
